docs/okeps/okep-5224-connecting-udns/okep-5224-connecting-udns.md

```diff
@@ -0,0 +1,543 @@
+# OKEP-5224: Connecting UserDefinedNetworks
+
+## Problem Statement
+
+The problem we are trying to solve here is how can an admin connect
+multiple [UserDefinedNetworks] or [ClusterUserDefinedNetworks] better known
+as “Connecting (C)UDNs”. The assumption here is that each (C)UDN is an
+isolated island and in order to make any two (C)UDNs communicate with
+each other they have to be explicitly requested to be connected.
+
+[UserDefinedNetworks]: https://ovn-kubernetes.io/api-reference/userdefinednetwork-api-spec/#userdefinednetwork
+[ClusterUserDefinedNetworks]: https://ovn-kubernetes.io/api-reference/userdefinednetwork-api-spec/#clusteruserdefinednetwork
+
+## Goals
+
+* Connecting and disconnecting UDNs and CUDNs within the same cluster
+  aka intra-cluster-udn-connectivity for admins (can be done as day0 or day2).
+  * Support for connecting `Layer3<->Layer3`, `Layer2<->Layer2` and `Layer3<->Layer2` type networks.
+  * Support for both `role:Primary` and `role:Secondary` CUDN connections.
+  * Support for both `role:Primary` and `role:Secondary` NAD connections. From
+    henceforth all references to CUDN/UDNs implies NADs as well.
+  * Support for both `role:Primary` UDN connections.
+* Support for pods, services (clusterIPs only since expectation today is that
+  NodePorts and LoadBalancer type services are already exposed externally and
+  hence reachable across UDNs) and network policies across the connected UDNs and CUDNs.
+* Ensure the proposed API is expandable for also connecting and disconnecting
+  UDNs or CUDNs across clusters in the future.
+
+## Future Goals
+
+* Tenants being able to request for interconnecting their networks
+  with other tenants.
+    * We need to understand exact use cases for why we need tenants to initiate
+      connection requests with other tenants. How will a tenant know other tenants
+      that exist on the cluster? In cases like finance tenant wanting to use IT
+      tenant's services, they could open a ticket to the admin to make that happen.
+* Admins being able to connect `role:Secondary` UDNs. Given
+  `SecondaryUserDefinedNetworkSelector` uses a combination of
+  namespace selector and network selector, a tenant could change
+  the labels on the UDN thereby causing security issues. Hence this
+  type of selector support will come later
+* Connecting and disconnecting UDNs and CUDNs across two clusters
```

Comment:
i'm not sure we are going to need this anymore for multi-cluster. With connecting UDNs between clusters the admin will simply  make EVPN config that maps the UDNs to the same VPN and then when they create their UDNs, they need to take care that there is no IP conflict/subnet overlap. Either way I think that lives outside of OVN kubernetes itself or this CRD. 

Maybe for now we should just put in non-goals: multi-cluster connectivity

Reply:
this was discussed in the sync meeting. we are not planning for multi-cluster UDNs currently in first iteration and current plan is to go with EVPN if users want to do this (but its also possible this design will change on RH side... we don't know, let's see) for now I can remove the multi-cluster scope from this OKEP to keep it simpler

Reply:
moved to non-goals